src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -156,36 +153,32 @@ class ContentSidebar extends PureComponent<Props, State> {
     }
 
     /**
-     * Fetches the root folder on load
+     * Fetches the file data on load
      *
      * @private
      * @inheritdoc
      * @return {void}
      */
     componentDidMount() {
-        this.fetchData(this.props);
+        this.fetchFile();
     }
 
     /**
-     * Called when sidebar gets new properties
+     * Fetches new file data on update
      *
      * @private
+     * @inheritdoc
      * @return {void}
      */
-    componentWillReceiveProps(nextProps: Props): void {
+    componentDidUpdate(prevProps: Props): void {
         const { fileId, isLarge }: Props = this.props;
-        const { file, editors, hasBeenToggled }: State = this.state;
-        const hasVisibilityChanged = nextProps.isLarge !== isLarge;
-        const hasFileIdChanged = nextProps.fileId !== fileId;
-
-        if (hasFileIdChanged) {
-            // Clear out existing state
-            this.setState({ ...this.initialState });
-            this.fetchData(nextProps);
-        } else if (!hasBeenToggled && hasVisibilityChanged) {
-            this.setState({
-                view: this.getDefaultSidebarView(nextProps, file, editors),
-            });
+        const { fileId: prevFileId, isLarge: prevIsLarge }: Props = prevProps;
+        const { view }: State = this.state;
+
+        if (fileId !== prevFileId) {
```

Comment:
Any reason why you're removing the variables that were here before?

Reply:
Mainly that it's a different function, I think. Do you see any issues with it?

Reply:
nope, just a difference in style. Fine by me